### ISSUE TITLE
{:array, :map} type can't be inserted via Ash action

### ISSUE THREAD

**Issue by hanrelan** [ash-project/ash_postgres#212]
**Describe the bug**
This code in a resource:

```elixir
    attribute :speech_timestamps, {:array, :map}, allow_nil?: true, writable?: true
[...]
    update :set_speech_timestamps do
      accept [:speech_timestamps]
    end
[...]
```
results in this error when I call `Resource.set_speech_timestamps!`:
```
* ** (Ecto.QueryError) unsupported expression: %{start: 0.0, end: 183.552} in query:
from a0 in Scribble.Scribe.AudioFile,
  as: 0,
  where: type(a0.id, {:parameterized, Ash.Type.UUID.EctoType, []}) ==
  type(^..., {:parameterized, Ash.Type.UUID.EctoType, []}),
  update: [
  set: [
    speech_timestamps: [%{start: 0.0, end: 183.552}],
...
```

**Expected behavior**
I'd expect this to insert into the database.

** Runtime
 - Elixir version: 1.15.7
 - Erlang version: OTP 26
 - OS: Mac
 - Ash version: 
  :ash, "2.19.5"
 - any related extension versions:
 :ash_postgres, "1.5.7"
**Additional context**
This used to work with Ash 2.18 and ash_postgres 1.4.0

**Comment by zachdaniel** [comment]
Got this reproduced, it's an issue in `ash_postgres`. Going to be a tricky one to figure out :)

**Comment by zachdaniel** [comment]
Fixed in v1.5.8
